src/activity_item_handling.cpp

```diff
@@ -2676,6 +2676,15 @@ static bool generic_multi_activity_check_requirement( player &p, const activity_
                     candidates.push_back( point_elem );
                 }
                 if( candidates.empty() ) {
+                    if( p.is_npc() ) {
```

Comment:
These two checks are the same. If `is_npc` returns true, it should be convertible to a `npc*`, so you don't need to check the result.

Anyway, I suggest this simple line:
```suggestion
                    if( npc *const guy = dynamic_cast<npc *>( &p ) ) {
                        const std::string text = ...
```